sway-error/src/error.rs

```diff
@@ -144,11 +146,19 @@ pub enum CompileError {
         interface_name: InterfaceName,
         span: Span,
     },
+    #[error("Type \"{name}\" is not a part of {interface_name}'s interface surface.")]
+    TypeNotAPartOfInterfaceSurface {
+        name: Ident,
+        interface_name: InterfaceName,
+        span: Span,
+    },
     #[error("Constants are missing from this trait implementation: {missing_constants}")]
     MissingInterfaceSurfaceConstants {
         missing_constants: String,
         span: Span,
     },
+    #[error("Associated types are missing from this trait implementation: {missing_types}")]
+    MissingInterfaceSurfaceTypes { missing_types: String, span: Span },
```

Comment:
I know other `Missing`whatever errors do the formatting of the `String` attribute on the creation side but that's a leaky abstraction. Could we refactor this one (and the others), to contain a `Vec` and only do the formatting on the rendering side?

Reply:
The Missing errors now receive a `Vec<BaseIdent>` and moved the formatting to the rendering side.